app/assets/stylesheets/questions.css.scss

```diff
@@ -1,48 +1,56 @@
-.all-mine-other-filter {
-  margin-right: 25px;  // aligns with right side of question list
-}
-
-// (All-mine-filter relocation at different screen sizes)
-// On small screens (bootstrap xs) show all-mine-other filter
-// just above the question list, since it wraps onto its own .row
-#all-mine-filter-container-xs {
+// On small screens (bootstrap xs) show-small-screen and replace with show-large-screen otherwise
+#show-small-screen {
   display: none;
 }
 @media only screen and (max-width: 767px) {
-  #all-mine-filter-container-xs {
+  #show-small-screen {
     display: block;
   }
-  #all-mine-filter-container {
+  #show-large-screen {
     display: none;
   }
 }
-// End of (All-mine-filter relocation at different screen sizes)
 
-.generate-question-button-filter {
-  margin-left: 12px;  // aligns with left side of question list
-}
 
-// (generate-question-button-filter relocation at different screen sizes)
-// On small screens (bootstrap xs) show generate-question-button-filter
-// just above the question list, since it wraps onto its own .row
-#generate-question-button-filter-xs {
-  display: none;
-}
 @media only screen and (max-width: 767px) {
-  #generate-question-button-filter-xs {
-    display: block;
+  
+  #all-mine-other-group {
+    float: none !important;
+    text-align: center;
   }
-  #generate-question-button-filter {
-    display: none;
+
+  #generate-question-offset {
+    width: 100%;
+
+    a {
+      display: block;
+    }
   }
+
+}
+
+#all-mine-other-offset {
+  padding: 20px 25px 10px 0px;
+}
+
+#all-mine-other-group {
+  float: right;
 }
-// End of (generate-question-button-filter relocation at different screen sizes)
+
+#generate-question-offset {
```

Comment:
I have a few preferences with respect to the ID names. Generally an ID is unique on a page and should be used to identify the element itself (semantically), vs. specifying behavior. Usually when behaviors are the important thing one can use a class tag -- these are not required to be unique (so they're reusable), and by convention a class added to an element clues the developer reading the source that the element has some customized behavior or styling (he/she can then go look in the CSS files to find out how it should behave).

All of the `*-offset` and `show-*` IDs here do look like they're unique, but the IDs aren't semantically useful (I don't know what to expect inside the `#show-small-screen` element -- if it were a `.show-small-screen` class, I would easily be able to ascertain the behavior though). A suggestion would be to split the difference here: 

* make the `show-*` into classes, not IDs (that way you could theoretically use them on other elements to hide/show, and don't need to change the semantic markup of the IDs at all)
* keep the `*-offset`s as IDs, but make the names more semantically helpful, like `#all-mine-other-container` or `#all-mine-other-col` or something. (The idea is that in this case the padding is so specific to that exact element that making it a class wouldn't be very helpful.)